nixos/modules/system/boot/systemd.nix

```diff
@@ -52,7 +52,7 @@ let
       "systemd-udevd-kernel.socket"
       "systemd-udevd.service"
       "systemd-udev-settle.service"
-      ] ++ (optional (!config.boot.isContainer) "systemd-udev-trigger.service") ++ [
+      ] ++ (optional (!config.boot.isContainer || config.virtualisation.lxc.nestedContainer) "systemd-udev-trigger.service") ++ [
```

Comment:
would this also apply to privleged containers or do they not have this problem?

Reply:
I'll have to test this. I never use privileged containers.

Reply:
I've now tested adding a condition for `config.virtualisation.lxc.privilegedContainer` and this change is not sufficient to start this service. Apparently `/sys` is mounted read-only in a privileged container and this fails the condition checks for `systemd-udev-trigger`.

```
[root@test1:~]# systemctl status systemd-udev-trigger
○ systemd-udev-trigger.service - Coldplug All udev Devices
     Loaded: loaded (/etc/systemd/system/systemd-udev-trigger.service; enabled; preset: enable>
    Drop-In: /nix/store/35b21hffqfwv64h5hnlavza49zi6xjg4-system-units/service.d
             └─zzz-lxc-service.conf
     Active: inactive (dead)
  Condition: start condition failed at Wed 2023-09-27 09:30:05 EDT; 11s ago
             └─ ConditionPathIsReadWrite=/sys was not met
       Docs: man:udev(7)
             man:systemd-udevd.service(8)
```


Fortunately for privileged container users, this service isn't completely necessary as networkd does to correctly configure the interface for either static or dhcp. The only impact I can see from `systemd-udev-trigger` failing to start is that `networkctl status eth0` is unable to detect the link file, but it *is* able to locate the network file.

```
● 32: eth0                                                                     
                     Link File: n/a
                  Network File: /etc/systemd/network/40-eth0.network
                         State: routable (configured)
```

Since this works as-is, and adding the condition here would not fix things further, I will not change the PR.

Thanks for the suggestion though. :)